extensions/ql-vscode/test/pure-tests/remote-queries/gh-api/gh-api-client.test.ts

```diff
@@ -0,0 +1,93 @@
+import { expect } from 'chai';
+import * as path from 'path';
+
+import * as Octokit from '@octokit/rest';
+import { retry } from '@octokit/plugin-retry';
+
+import { setupServer } from 'msw/node';
+
+import {
+  getRepositoryFromNwo,
+  getVariantAnalysis,
+  getVariantAnalysisRepo, getVariantAnalysisRepoResult,
+  submitVariantAnalysis
+} from '../../../../src/remote-queries/gh-api/gh-api-client';
+import { Credentials } from '../../../../src/authentication';
+import {
+  createMockSubmission
+} from '../../../../src/vscode-tests/factories/remote-queries/shared/variant-analysis-submission';
+import { createRequestHandlers } from '../../../../src/mocks/request-handlers';
+
+const mockCredentials = {
+  getOctokit: () => Promise.resolve(new Octokit.Octokit({ retry }))
+} as unknown as Credentials;
+
+const server = setupServer();
+
+before(() => server.listen());
+
+afterEach(() => server.resetHandlers());
+
+after(() => server.close());
+
+async function loadScenario(scenarioName: string) {
```

Comment:
This and `setupServer` feel like they belong in `MockGitHubApiServer`. Do we need to refactor a bit to allow easy use of `MockGitHubApiServer` in automated tests? Maybe we need to separate the core server functionality with the extension interactions?

Reply:
Thanks for refactoring!